src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -504,8 +507,14 @@ internal static ODataUriResolver GetUriResolver(IServiceProvider container)
             return container.GetRequiredService<ODataUriResolver>();
         }
 
-        private static IList<T> FindAcrossModels<T>(IEdmModel model, String qualifiedName, bool caseInsensitive) where T : IEdmSchemaElement
+        private static IReadOnlyList<T> FindAcrossModels<T>(IEdmModel model, String qualifiedName, bool caseInsensitive) where T : IEdmSchemaElement

```

Comment:
Beyond the scope of this PR: the private methods `FindAcrossModels` and `FindSchemaElements` don't seem to **need** to return an IList: all their callers pretty much use the return value as an IEnumerable.
As a result, the List allocation could probably be avoided (including in the new CacheInsensitiveSchemaElementsCache.FindElementsOfType).

Reply:
There are methods which index into the results, e.g. `ResolveBoundOperations`. But they do the index lookup in a plan for loop, so this could be achieved with a `foreach` on an `IEnumerable`. But I assume a for loop was chosen to avoid the cost of `foreach` against an `IEnumerable`.